src/components/reusable-ui/Card.js

```diff
@@ -1,22 +1,32 @@
-import styled from "styled-components";
+import styled, { css } from "styled-components";
 import { theme } from "../../theme";
 import Button from "./Button";
 import { TiDelete } from "react-icons/ti"
 
-export default function Products({ title, imageSource, leftDescription, hasDeleteButton, onDelete }) {
+export default function Products({ title, imageSource, leftDescription, hasDeleteButton, onDelete, onClick, isHoverable, isSelected }) {
     return (
-      <CardStyled>
-        { hasDeleteButton && <button className="delete-button" aria-label="delete-button" onClick={onDelete}><TiDelete className="icon"/></button> }
-        
-        <div className="product-img">
-          <img src={imageSource} alt={title} />
-        </div>
-        <div className="text-info">
-          <div className="title">{title}</div>
-          <div className="description">
-            <div className="left-description">{leftDescription}</div>
-            <div className="right-description">
-              <Button className="primary-button" label={"Ajouter"} />
+      <CardStyled
+        onClick={onClick}
+        isHoverable={isHoverable}
+        isSelected={isSelected}
+      >
+        <div className="card">
+          { hasDeleteButton && <button className="delete-button" aria-label="delete-button" onClick={onDelete}><TiDelete className="icon"/></button> }
+          
+          <div className="product-img">
+            <img src={imageSource} alt={title} />
+          </div>
+          <div className="text-info">
+            <div className="title">{title}</div>
+            <div className="description">
+              <div className="left-description">{leftDescription}</div>
+              <div className="right-description">
+                <Button
+                  className="primary-button"
+                  label={"Ajouter"}
+                  onClick={(event) => event.stopPropagation()}
```

Comment:
```suggestion
                  onClick={onClick}
```

et ensuite, le onClick que tu envoies en valeur, il te vient des props de Card.js